### ISSUE TITLE
Deferred jobs

### ISSUE THREAD

**Issue by scriby** [scttnlsn/monq#20]
The use case I have looks something like this:
- Only one job in a category can run at once
- Jobs within a category must be processed in order
- Jobs within different categories should run in parallel
- The categories are dynamic and manifest themselves at runtime

The simplest way I can see to solve this is when picking up a job, check whether there is already a job in progress or deferred within the category. If so, defer the current job. Each time a job finishes (or errors), un-defer the next job in the category.

Is this something you'd be ok taking on in monq? Ok if I implement and send a PR?
